src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2043,11 +2044,24 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             }
 
             // iterate over the collection asking the engine for the name
-            foreach (var pair in threads)
+            foreach ((int id, IDebugThread2 thread) in threads)
             {
-                string name;
-                pair.Value.GetName(out name);
-                response.Threads.Add(new OpenDebugThread(pair.Key, name));
+                string name = null;
+
+                if (thread != null && thread.GetName(out name) == HRConstants.S_OK)
```

Comment:
>thread != nul [](http://example.com/codeflow?start=20&length=13)

I don't know how thread could ever be null (probably couldn't be), but if we want to handle that case, we should just ignore it rather than adding it to the collection.